### PR TITLE
I2C exercises: bullet points for content

### PR DESCRIPTION
**Pull request by Dajamante**
Points to be added to the driver exercise 
NEED TO ADD BAUD RATE

### CODE REVIEW

File: book/src/04_3_3_i2c.md [hunk 1]
```diff
@@ -1 +1,187 @@
...
+
+
+🔎  We tell the device which one we want it to use by applying either 0V or 3.3V to the AP_AD0 pin on the device. If we apply 0V, it listens to address 0x68. If we apply 3.3V it listens to address 0x69. You can therefore think of pin AD_AD0 as being a one-bit input which sets the least-significant bit of the device address.
+
+
+ ✅ Create an enum that represents both address variants. The values of the variants need to be in binary representation. 
+
+<Details>
+    <Summary>Example Solution</Summary>
+```rust
```
Comment (BriocheBerlin) [inline book/src/04_3_3_i2c.md:74]: Following all the formatting of the examples is broken. It seems doing `\```rust` is not necessary when 
`\<Details>` and `\<Summary>` is used.

File: book/src/04_3_3_i2c.md [hunk 2]
```diff
@@ -1 +1,187 @@
...
+impl<I2C, E>ICM42670P<I2C>
+where
+    I2C: i2c::WriteRead<Error = E> + i2c::Write<Error = E>,
+{
+    /// Create a new instance of the ICM42670P.
+    pub fn new(i2c: I2C, address: DeviceAddr) -> Result<Self, E> {
+
+        let icm42670p = ICM42670P { i2c, address };
```
Comment (BriocheBerlin) [inline book/src/04_3_3_i2c.md:44]: It looks like this in `lib.rs`. Here shouldn't it be only `Ok(Self { i2c, address })`?

File: book/src/04_3_3_i2c.md [hunk 3]
```diff
@@ -1 +1,187 @@
...
+    * switching the the gyroscope sensor or the accelerometer on
+    * starting measurements
+    * reading measurements
+
+#### General info about how registers work
+
+- Registers are small amounts of storage, immediately accessible by the processor. The registers on the sensor are 8 bits.
+- They can be accessed by their address
+- You can find [register maps](https://3cfeqx1hf82y3xcoull08ihx-wpengine.netdna-ssl.com/wp-content/uploads/2021/07/DS-000451-ICM-42670-P-v1.0.pdf) in the section 14.
+- Returning a value with MSB and LSB is done by shifting MSB values, and OR LSB values.
```
Comment (BriocheBerlin) [inline book/src/04_3_3_i2c.md:181]: Same as in `easy`: what is MSB and LSB?

File: book/src/04_3_3_i2c.md [hunk 4]
```diff
@@ -1 +1,187 @@
...
+```rust
+let GYRO_DATA_X: i16 = ((GYRO_DATA_X1 as i16) << 8) | GYRO_DATA_X0 as i16;
+```
+(maybe from `from_be_bytes[GYRO_data_X1, GYRO_DATA_X2] is better)
```
Comment (BriocheBerlin) [inline book/src/04_3_3_i2c.md:185]: Same as in `easy`: this seems like a note to self.

**Review by BriocheBerlin — CHANGES_REQUESTED** [review]
Probably the whole solution folder needs to go. Also it seems that in the `hard` exercise most formatting is broken.

**Comment by Mirabellensaft** [comment]
I have removed all the hidden code in the hard version because there was no clean and easy way to do it. People just have to check in the corresponding section in the easy version. This also removes tons of duplicate information \o/. I have found a way to fix this for newer mdbook versions though.

**Review by BriocheBerlin — APPROVED** [review]

**Comment by BriocheBerlin** [comment]
Not sure why Docker fails... Otherwise looks good 👍

**Pull request merged**